### PR TITLE
moved template logic out of dashboards sync script to helm

### PR DESCRIPTION
**Pull request by AndrewChubatiuk**
Moved part of templating logic out of sync script to Helm template

### CODE REVIEW

File: charts/victoria-metrics-k8s-stack/templates/grafana/dashboard.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+{{- /*
+Check if dashboard is enabled
+*/}}
+{{- if $dashboardCondition }}
+
+{{- $dashboardResult := toRawJson $dashboardData -}}
+
+{{- if $.Values.grafanaOperatorDashboardsFormat.enabled }}
+apiVersion: grafana.integreatly.org/v1beta1
```
Comment (Haleygo) [inline charts/victoria-metrics-k8s-stack/templates/grafana/dashboard.yaml:42]:
```suggestion
{{- if $.Values.grafanaOperatorDashboardsFormat.enabled }}
---
apiVersion: grafana.integreatly.org/v1beta1
```

File: charts/victoria-metrics-k8s-stack/templates/grafana/dashboard.yaml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  {{- with $.Values.grafanaOperatorDashboardsFormat.instanceSelector }}
+  instanceSelector:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  {{- if $.Values.grafanaOperatorDashboardsFormat.allowCrossNamespaceImport }}
+  allowCrossNamespaceImport: true
+  {{- end }}
+  json: | {{ $dashboardResult | nindent 4 }}
+{{- else }}
+apiVersion: v1
```
Comment (Haleygo) [inline charts/victoria-metrics-k8s-stack/templates/grafana/dashboard.yaml:60]:
```suggestion
{{- else }}
---
apiVersion: v1
```

File: charts/victoria-metrics-k8s-stack/templates/grafana/dashboard.yaml (hunk 1 above)
Comment (AndrewChubatiuk) [reply charts/victoria-metrics-k8s-stack/templates/grafana/dashboard.yaml]: updated

File: charts/victoria-metrics-k8s-stack/templates/grafana/dashboard.yaml (hunk 2 above)
Comment (AndrewChubatiuk) [reply charts/victoria-metrics-k8s-stack/templates/grafana/dashboard.yaml]: updated

File: charts/victoria-metrics-k8s-stack/hack/sync_grafana_dashboards.py [hunk 3]
```diff
@@ -86,246 +63,148 @@ def new_representer(dumper, data):
...
+        target['expr'] = target['expr'].rstrip()
+        if '\n' in target['expr']:
+            target['expr'] = literal(target['expr'])
 
 def replace_ds_type_in_panel(panel):
+    if 'gridPos' in panel:
+        if 'y' in panel['gridPos']:
+            y = panel['gridPos']['y']
+            del panel['gridPos']['y']
+            panel['gridPos']['[[ .yaxis ]]'] = y
```
Comment (Haleygo) [inline charts/victoria-metrics-k8s-stack/hack/sync_grafana_dashboards.py:104]: Could you help adding some comments here, to help understanding the issue for future reader?

**Review by Haleygo — APPROVED** [review]
LGTM

File: charts/victoria-metrics-k8s-stack/hack/sync_grafana_dashboards.py (hunk 3 above)
Comment (AndrewChubatiuk) [reply charts/victoria-metrics-k8s-stack/hack/sync_grafana_dashboards.py:106]: done

**Review by Haleygo — APPROVED** [review]
Thank you!

**Pull request merged**